Public/Src/Cache/ContentStore/Hashing/ByteArrayComparer.cs

```diff
@@ -217,9 +203,68 @@ private static int CompareArrays(IList<byte> x, IList<byte> y)
         ///     Static comparison method for when the caller does not want to create an instance
         ///     of this class.
         /// </summary>
-        public static bool ArraysEqual(IList<byte> x, IList<byte> y)
+        public static unsafe bool ArraysEqual(byte[] x, byte[] y)
         {
-            return CompareArrays(x, y) == 0;
+            // Adapted from: https://gist.github.com/airbreather/90c5fd3ba9d77fcd7c106db3beeb569b
+            // (https://stackoverflow.com/questions/43289/comparing-two-byte-arrays-in-net, Joe Amenta's answer).
+            // TODO: Once we get to NetCore 2.2+, consider ReadOnlyMemory<byte>.SequenceEqual which should be

```

Comment:
> get to NetCore 2.2+, [](start = 28, length = 21)

We are actually compiling this for:
* netstandard20
* net451
* net461
* net472
* netcoreapp30

BuildXL is slowly moving away from net472 and moving to netcoreapp30. We are close to flipping it in Cloudbuild.

Do you mind measuring on netcore30 and if not improved their, `#if FEATURE_CORECLR`

Reply:
Actually since we are on notecore30 the SO thread refers to SIMD. Should definately be worth trying

---
In reply to: [307537497](https://github.com/microsoft/BuildXL/pull/662#discussion_r307537497) [](ancestors = 307537497)

Reply:
I have a PR #665 to stop building this for net461 (just FYI)

Reply:
The PR went in.

Reply:
Significant speedup with Span.SequenceEqual, added a split in the implementation based on netcore vs. framework.